core/src/main/kotlin/io/specmatic/core/Feature.kt

```diff
@@ -538,16 +538,11 @@ data class Feature(
 
     private fun getScenarioWithDescription(scenarioResult: ReturnValue<Scenario>): ReturnValue<Scenario> {
         return scenarioResult.ifHasValue { result: HasValue<Scenario> ->
-            val descriptionFromPlugin = result.value.descriptionFromPlugin?.takeIf {
-                it.isNotBlank()
-            }?.plus(" ") ?: ""
-            val description = result.valueDetails.singleLineDescription()
-
-            val tag = if(description.isNotBlank())
-                " [${description}]"
-            else
-                ""
-            HasValue(result.value.copy(descriptionFromPlugin = "$descriptionFromPlugin${result.value.apiDescription}$tag"))
+            val apiDescription = result.value.descriptionFromPlugin ?: result.value.apiDescription
+            val tag = result.valueDetails.singleLineDescription().let {
+                if (it.isNotBlank()) " [$it]" else ""
+            }
+            HasValue(result.value.copy(descriptionFromPlugin = "$apiDescription$tag"))
```

Comment:
This is not a good idea. Use .ifHasValue (here you seem to need the result).. ReturnValues may also have failures or Exceptions, and calling .value will result in an exception thrown in those cases.

Reply:
Hey Joel, I am already using the .ifHasValue on line 540. I think that should suffice. But I may be wrong. Please let me know if it does not suffice.